frontend/src/metabase/public/LocaleProvider.tsx

```diff
@@ -3,27 +3,42 @@ import { type PropsWithChildren, useEffect, useState } from "react";
 import { setLocaleHeader } from "metabase/lib/api";
 import { loadLocalization, setUserLocale } from "metabase/lib/i18n";
 
+interface LocaleProviderProps {
+  locale?: string | null;
+  shouldWaitForLocale?: boolean;
+}
+
 export const LocaleProvider = ({
   children,
   locale,
-}: PropsWithChildren<{ locale?: string | null }>) => {
+  shouldWaitForLocale,
+}: PropsWithChildren<LocaleProviderProps>) => {
   // The state is not used explicitly, but we need to trigger a re-render when the locale changes
   // as changing the locale in ttag doesn't trigger react components to update
-  const [_isLoadingLocale, setIsLoadingLocale] = useState(false);
+  const [isLocaleLoaded, setIsLocaleLoaded] = useState(false);
 
   useEffect(() => {
     if (locale) {
-      setIsLoadingLocale(true);
       setLocaleHeader(locale);
-      loadLocalization(locale).then(translatedObject => {
-        setIsLoadingLocale(false);
-        setUserLocale(translatedObject);
-      });
+      loadLocalization(locale)
+        .then(translatedObject => {
+          setIsLocaleLoaded(true);
+          setUserLocale(translatedObject);
+        })
+        .catch(() => {
+          setIsLocaleLoaded(true);
+        });
```

Comment:
since the new code guards against the passed `locale`, we need to deal with failure as well when the locale fails to load for whatever reason e.g. incorrect locale.

Reply:
should we switch the state to be "isLocaleLoading"? I think it's more standard and otherwise in case of error `isLocaleLoaded` would be true even though technically it's not loaded
we could do `useState(Boolean(locale))` to ensure that the first render is with the correct value.

Reply:
I think it would also improve the logic for the return, as it will be `if(isLocaleLoading && shouldWaitForLocale) return null` (or a loader?)

Reply:
The problem with `isLoading` is that by default it's `false`, so the state goes from
- `false` (initial) -> `true` (loading) -> `false` (loaded)

So it's hard to distinguish between the initial state and the loaded state.

Reply:
Yeah that's why I suggested doing `useState(Boolean(locale))`, so the initial value would be correct no?

Reply:
Fixed in c3348b22a846b81fb5db246ce468226390e2ff29. I also ensure the condition used to load the locale and set the state is the same one and clean up some comments.